editor/editor_help_search.cpp

```diff
@@ -117,8 +117,12 @@ void EditorHelpSearch::_notification(int p_what) {
 			_update_icons();
 		} break;
 
-		case NOTIFICATION_ENTER_TREE: {
+		case NOTIFICATION_READY: {
 			connect("confirmed", callable_mp(this, &EditorHelpSearch::_confirmed));
+		} break;
+
+		case NOTIFICATION_THEME_CHANGED:
+		case NOTIFICATION_ENTER_TREE: {
```

Comment:
`NOTIFICATION_THEME_CHANGED` is now received automatically on enter tree, so this is no longer necessary.

Reply:
Are you sure about this? I remembered there was a PR ensuring this, but IIRC it introduces regressions and was reverted.

Reply:
https://github.com/godotengine/godot/blob/e0e93ce094d19d132f34aa5c92e892524a8125c9/scene/main/window.cpp#L1155

Reply:
Can confirm that works without that line as suggested.

Reply:
Then you can revert this whole file, no? The connection can be done in enter tree as before.

Reply:
No, because it may enter tree multiple times and `connect` print an error when the signal is already connected.

Reply:
This should be `NOTIFICATION_THEME_CHANGED` if the only thing it does is updating icons.
What I meant with the first comment is that this notification is received when entering tree. But it can also be received afterwards when the theme changes, so ENTER_TREE is not sufficient here.

Reply:
I can do that, but do note that this PR only moves the `connect` call into NOTIFICATION_READY.

Reply:
I know, but now the theme changed notification can be used here.